### PR TITLE
Adding multiple heatmap styling example

### PR DESCRIPTION
**Pull request by langsmith**
This pr adds an example which shows off various styling options for 🔥🗺 

ezgif com-resize 1
ezgif com-optimize 2

### CODE REVIEW

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+    initHeatmapRadiusStops();
+    initHeatmapIntensityStops();
+    addHeatmapLayer();
+    findViewById(R.id.switch_heatmap_style_fab).setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        if (index == listOfHeatmapColors.length - 1) {
+          index = 0;
+        }
+        if (firstLoad) {
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java]: why not remove this boolean and move the `index++` to the top of this method?

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+          index = 1;
+          firstLoad = false;
+        }
+        mapboxMap.getLayer(HEATMAP_LAYER_ID).setProperties(
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java]: getLayer could return null?

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+  private void addHeatmapDataSource() {
+    mapboxMap.addSource(new GeoJsonSource(HEATMAP_SOURCE_ID,
+      loadGeoJsonFromAsset("la_heatmap_styling_points.geojson")));
+  }
+
+  private void addHeatmapLayer() {
+    // Create the heatmap layer
+    HeatmapLayer layer = new HeatmapLayer(HEATMAP_LAYER_ID, HEATMAP_SOURCE_ID);
+
+    // Heatmap layer disappears at whatever zoom level is set as the maximum
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java]: Love the commenting here.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+        literal(0.3), rgba(0, 74, 255, .2),
+        literal(0.4), rgba(0, 202, 255, .25),
+        literal(0.5), rgba(73, 255, 154, .3),
+        literal(0.6), rgba(171, 255, 59, .35),
+        literal(0.7), rgba(255, 197, 3, .4),
+        literal(0.8), rgba(255, 82, 1, 0.7),
+        literal(0.9), rgba(196, 0, 1, 0.8),
+        literal(0.95), rgba(121, 0, 0, 0.8)
+      ),
+      interpolate(
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java]: `// 11`

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  private void initHeatmapColors() {
+    listOfHeatmapColors = new Expression[] {
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java]: You numbered some of these but not all of them. Make it consistent?

File: MapboxAndroidDemo/src/main/res/layout/activity_multiple_heatmap_styling.xml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        android:layout_height="wrap_content"
+        android:layout_gravity="bottom|right"
+        android:layout_marginBottom="16dp"
+        android:layout_marginEnd="16dp"
+        android:layout_marginRight="16dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:srcCompat="@drawable/ic_swap_horiz_white_24dp"
+        mapbox:fabSize="normal" />
+
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/res/layout/activity_multiple_heatmap_styling.xml]: nit

File: MapboxAndroidDemo/src/main/res/layout/activity_multiple_heatmap_styling.xml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    xmlns:mapbox="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <com.mapbox.mapboxsdk.maps.MapView
+        android:id="@id/mapView"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        mapbox:mapbox_styleUrl="@string/mapbox_style_light" />
+
```
Comment (cammace) [inline MapboxAndroidDemo/src/main/res/layout/activity_multiple_heatmap_styling.xml]: nit

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java (hunk 2 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java]: yes, conceivably. Will add a null check

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java (hunk 5 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java]: Added numbers for all arrays

File: MapboxAndroidDemo/src/main/res/layout/activity_multiple_heatmap_styling.xml (hunk 6 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/res/layout/activity_multiple_heatmap_styling.xml]: Fixed

File: MapboxAndroidDemo/src/main/res/layout/activity_multiple_heatmap_styling.xml (hunk 7 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/res/layout/activity_multiple_heatmap_styling.xml]: Fixed

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java (hunk 1 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java]: both done

**Pull request merged**